File: backend/app/schemas/ingredient.py
"""Ingredient schemas."""

from datetime import datetime
from decimal import Decimal

from pydantic import BaseModel, ConfigDict, Field

from app.models.ingredient import Unit


class IngredientCreate(BaseModel):
    name: str = Field(min_length=1, max_length=120)
    unit: Unit = Unit.GRAM
    shelf_life_days: int | None = Field(default=None, ge=0)


class IngredientUpdate(BaseModel):
    name: str | None = Field(default=None, min_length=1, max_length=120)
    unit: Unit | None = None
    shelf_life_days: int | None = Field(default=None, ge=0)
    is_active: bool | None = None


class IngredientRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    name: str
    unit: Unit
    shelf_life_days: int | None
    is_active: bool
    created_at: datetime
    updated_at: datetime


class IngredientWithStockRead(IngredientRead):
    """Ingredient + its current stock_level (single row, may be None)."""

    quantity: Decimal | None = None
    reorder_point: Decimal | None = None

File: frontend/src/types/ingredient.ts
/** Measurement unit for an ingredient (mirrors backend Unit enum values). */
export type Unit = 'g' | 'kg' | 'ml' | 'l' | 'piece' | 'shot' | 'pump' | 'pack'

export const UNITS: Unit[] = ['g', 'kg', 'ml', 'l', 'piece', 'shot', 'pump', 'pack']

export const UNIT_LABELS: Record<Unit, string> = {
  g: 'กรัม (g)',
  kg: 'กิโลกรัม (kg)',
  ml: 'มิลลิลิตร (ml)',
  l: 'ลิตร (l)',
  piece: 'ชิ้น',
  shot: 'ช็อต',
  pump: 'ปั๊ม',
  pack: 'แพ็ค',
}

export interface Ingredient {
  id: number
  name: string
  unit: Unit
  shelf_life_days: number | null
  is_active: boolean
  created_at: string
  updated_at: string
}

/** Ingredient + its current stock row (Decimal strings; null when no stock row). */
export interface IngredientWithStock extends Ingredient {
  quantity: string | null
  reorder_point: string | null
}
